### ISSUE TITLE
Bad name for module in custom collection

### ISSUE THREAD

**Issue by LvffY** [ansible-community/antsibull-docs#132]
With my team, we have a custom collection. In this collection we have custom modules. 

In particular, we have these directory layout : 

![image](https://user-images.githubusercontent.com/37327656/233773499-32bf17e5-6107-4301-8e2b-89a386579007.png)

These modules must be called in our playbook with the entire directory names. For example, module **plugins/modules/cde/job.py** is called with **namespace.collection_name.<i>cde</i>.job**.

We're also building our documentation through **antsibull-docs**. 

Doing so we have a website with all our module documentation. However the name if the module name on the docs is not correctly set : in particular, we miss the intermediate directory name. For example, module **plugins/modules/cde/job.py** is referenced under **namespace.collection_name.job** (so without the ***cde*** directory). 

![image](https://user-images.githubusercontent.com/37327656/233773810-7570ab0c-b9fb-450c-bdb0-cb3c51b97c11.png)

In conclusion, I think that there is an issue when the docs is created : it looks like it removes all intermediate directories in the module names.

**Comment by felixfontein** [comment]
Yeah, this is a left-over from the flatmapping we needed for community.general and community.network. I think it is time to disable that and properly support long FQCNs.

Since this is a kind of breaking change it will have to wait for 2.0.0, which should be done in a couple of weeks at most (#104).

**Comment by LvffY** [comment]
> Yeah, this is a left-over from the flatmapping we needed for community.general and community.network. I think it is time to disable that and properly support long FQCNs.
> 
> 
> 
> Since this is a kind of breaking change it will have to wait for 2.0.0, which should be done in a couple of weeks at most (#104).

Seems legit to me !

Do we already know which breaking changes will occur in 2.0.0 ?

**Comment by felixfontein** [comment]
It's mainly dropping support for ansible-core 2.12 and earlier, and drop support for Python before 3.9. Besides that it's mainly internal changes / refactorings that shouldn't affect users.

**Comment by felixfontein** [comment]
#134 disables flatmapping by default.
